python/tchannel/tornado/tchannel.py

```diff
@@ -99,15 +103,38 @@ def __init__(self, hostport=None, process_name=None, known_peers=None):
             sys.argv[0], os.getpid()
         )
 
+        self.caller_name = caller_name
+
         # register event hooks
         self.event_emitter = EventEmitter()
         self.hooks = EventRegistrar(self.event_emitter)
-        self.hooks.register(ZipkinTraceHook(tchannel=self))
 
         if known_peers:
             for peer_hostport in known_peers:
                 self.peers.add(peer_hostport)
 
+    def advertise(self, caller_name, router):
+        """Advertise the given TChannel to Hyperbahn using the given name.
+
+        This informs Hyperbahn that the given service is hosted at this
+        TChannel at a fixed rate.
+
+        It also tells the TChannel about the given Hyperbahn routers.
+
+        :param tchannel:
+            TChannel to register with Hyperbahn
+        :param caller_name:
+            Name of the caller behind this TChannel
+        :param routers:
+            Seed list of addresses of Hyperbahn routers
+        :returns:
+            A future that resolves to the remote server's response after
+            the first advertise is successful.
+        """
+
+        self.caller_name = caller_name
```

Comment:
the caller name and the name you advertise as are different.

For example `rt-los_angeles` (advertise) vs `dispatch` (caller name).